program/rust/src/time_machine_types.rs

```diff
@@ -35,13 +368,20 @@ pub struct PriceAccountWrapper {
 }
 impl PriceAccountWrapper {
     pub fn initialize_time_machine(&mut self) -> Result<(), OracleError> {
-        msg!("implement me");
-        Ok(())
+        self.time_machine.initialize()
     }
 
     pub fn add_price_to_time_machine(&mut self) -> Result<(), OracleError> {
-        msg!("implement me");
-        Ok(())
+        //this would work on the first add_price as well, as the first entry is always invalid, so
+        // even though we would probabpy end up overflowing the first number that should not corrupt
+        // any valid data
+        self.time_machine.add_price(
```

Comment:
I think we need to be careful to only call this function when the price is successfully updated (i.e., the current status is TRADING). do we do that in the calling code?

Reply:
We only call it on upd_aggregate returns true, and it only does that when status is trading.
We can add an extra check to harden it if we want

Reply:
Can confirm this